compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -138,7 +138,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     #[inline]
     pub fn write_ty(&self, id: hir::HirId, ty: Ty<'tcx>) {
         debug!("write_ty({:?}, {:?}) in fcx {}", id, self.resolve_vars_if_possible(ty), self.tag());
-        self.typeck_results.borrow_mut().node_types_mut().insert(id, ty);
+        let mut typeck = self.typeck_results.borrow_mut();
+        let mut node_ty = typeck.node_types_mut();
+        if let Some(ty) = node_ty.remove(id)
+            && let ty::Error(e) = ty.kind()
+        {
+            // Do not overwrite nodes that were already marked as `{type error}`. This allows us to
+            // silence unnecessary errors from obligations that were set earlier than a type error
+            // was produced, but that is overwritten by later analysis.  This happens in particular
+            // for `Sized` obligations introduced in gather_locals.  (#117846)
+            node_ty.insert(id, ty);
```

Comment:
Why not `node_ty.get(id)` instead of `remove`+`insert`?

Reply:
[We don't have `LocalTableInContextMut::get`.](https://github.com/rust-lang/rust/blob/010f3944e0f0baac8d738da49772fd06acd3701b/compiler/rustc_middle/src/ty/typeck_results.rs#L565-L595)

Reply:
Wouldn't `get_mut` work here?

Reply:
It would. It's just that the type would get discarded and overwritten regardless of the success or not of the let chain, but yeah, I can make it use `get_mut`.

Reply:
Ah, my surprise came from the fact that you're reinserting the `ty` you just removed. I guess you intended to insert the `ty` you got as input to the function?

Reply:
I see. I would indeed prefer if you used `get_mut`

Reply:
Or implemented `get` tbh

Reply:
If you really prefer `remove`+`insert` could you then name the variable something like `err_ty` to clarify the intent?